lsaffre/sphinxfeed
# Unit test improvements

Closes #6.

* Test both RSS and Atom output formats
* Parse XML output and compare one element at a time. This gives better test output on failure, and ignores parts that don't matter.
* Report test status in CI. Failing tests will prevent merging a PR or publishing a package.
* Add coverage report config with `pytest-cov`. By default, it will show condensed terminal output (total coverage).  Adding `--cov-report=html` will generate a detailed HTML report (line coverage).
* Add some missing test coverage; now at 95%
* Run Sphinx builds in tests using `SphinxTestApp`.
    * This is a pytest plugin that comes with Sphinx, and is poorly documented but quite useful for testing. It does some extra environment setup that allows invoking Sphinx from python instead of `subprocess(['sphinx-build', ...])`.
    * You can see the full build output with `pytest -vs`.

Let me know if you have any suggestions, or questions about how it works!
